Update Semgrep Editor docs + add information re: structure mode

khorne3 opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

brandonspark commented on src/components/procedure/_enable-turbo-mode.md:
```diff
@@ -1,9 +1,9 @@
+Click the <i class="fa-solid fa-toggle-large-on"></i> **Turbo mode** toggle to enable.
 
-In this mode, the **Run** button is hidden because Semgrep scans upon detecting any change to the text. Turbo mode works in both Simple mode and Advanced mode. Turbo mode cannot be enabled for Semgrep Pro Engine.
+The **Run** button is hidden in Turbo mode because Semgrep scans after any changes to the rule. Turbo mode works with the editor in either simple or advanced mode.
```
```suggestion
The **Run** button is hidden in Turbo mode because Semgrep scans after any changes to the rule. Turbo mode works with the editor in either structure or advanced mode.
```

brandonspark replied:
(simple mode won't actually exist anymore)

brandonspark reviewed (approved):
looks good! it's worth noting that these screenshots are very subtly out of date (after I merged https://github.com/semgrep/semgrep-app/pull/13519), because `simple` mode will no longer exist, `structure` takes its place. but, it probably doesn't matter that this is 100% faithful. its great we're deprecating the really old playground screenshots :)

s-santillan commented:
Howdy!
1. Requesting that Pablo review these docs.
2. I think we'll want to remove the playground markdown file? Did we miss any information in the Playground that isn't in the Editor?

khorne3 commented:
@s-santillan Sounds good!

I don't think there's much difference in terms of the two, so I think we can delete playground and redirect?

pabloest commented:
@khorne3 Can you share some of the reasoning behind removing the Playground file? AFAICT structure mode is available in the Playground. Are we trying to instruct users to instead go to Semgrep Code > Editor for structure mode?

khorne3 commented:
@pabloest The playground doc has been hidden for some time now. I've moved info from the Playground doc to the Editor doc if necessary, and since people aren't supposed to see the former, I deleted it and added a redirect. Additionally, they're both so similar that one doc should be sufficient and make it easier to maintain

pabloest commented:
OK great! Thanks for the context, @khorne3.
